File: examples/javascript/streaming.mjs
// Streaming — tokens arrive in real-time with PII restored.
import OpenAI from 'openai';
import { createVeilOpenAIConfig } from 'a5omic-veil';

const openai = new OpenAI(createVeilOpenAIConfig({
  veilApiKey: process.env.VEIL_API_KEY,
  upstreamApiKey: process.env.OPENAI_API_KEY,
}));

const stream = await openai.chat.completions.create({
  model: 'gpt-4o-mini',
  messages: [{
    role: 'user',
    content: 'Write a formal reply to: I am Tom Ryan (tom@test.com), please help with my billing issue.'
  }],
  stream: true,
});

for await (const chunk of stream) {
  const content = chunk.choices[0]?.delta?.content;
  if (content) process.stdout.write(content);
}
console.log();

File: examples/python/streaming.py
"""Streaming example — tokens arrive in real-time with PII restored inline."""
import os
from openai import OpenAI

client = OpenAI(
    api_key=os.environ["OPENAI_API_KEY"],
    base_url="https://veil-api.com/v1",
    default_headers={
        "Authorization": f"Bearer {os.environ['VEIL_API_KEY']}",
        "x-upstream-key": os.environ["OPENAI_API_KEY"],
    },
)

stream = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {
            "role": "user",
            "content": "Write a formal reply to: Dear support, I am Tom Ryan (tom@example.com, 555-867-5309) and I need help with my account.",
        }
    ],
    stream=True,
)

for chunk in stream:
    content = chunk.choices[0].delta.content
    if content:
        print(content, end="", flush=True)

print()
